integration-cli/docker_cli_events_test.go

```diff
@@ -213,6 +213,44 @@ func TestEventsImageImport(t *testing.T) {
 	defer deleteAllContainers()
 	since := daemonTime(t).Unix()
 
+	finish := make(chan struct{})
+	defer close(finish)
+	id := make(chan string)
+	eventImport := make(chan struct{})
+
+	go func() {
+		eventsCmd := exec.Command(dockerBinary, "events", "--since", strconv.FormatInt(since, 10))
+		stdout, err := eventsCmd.StdoutPipe()
+		if err != nil {
+			t.Fatal(err)
+		}
+		err = eventsCmd.Start()
+		if err != nil {
+			t.Fatalf("failed to start 'docker events': %s", err)
+		}
+
+		go func() {
+			<-finish
+			eventsCmd.Process.Kill()
+		}()
+
+		containerID := <-id
+
+		matchImport := regexp.MustCompile(containerID + `: import$`)
+
+		scanner := bufio.NewScanner(stdout)
+		for scanner.Scan() {
+			if matchImport.MatchString(scanner.Text()) {
+				close(eventImport)
+			}
+		}
+
+		err = eventsCmd.Wait()
```

Comment:
I don't really undestand this, you mean declare eventsCmd outside of the goroutine?